tests/flags/positional_arguments.cc

```diff
@@ -0,0 +1,236 @@
+}
+
+TEST(PositionalArguments, BuildFileSucceed) {
+  test::ProcessFile msg;
+
+  auto parser = stout::flags::Parser::Builder(msg).Build();
+
+  std::array arguments = {
+      "program",
+      "build",
+      "--debug",
+      "foo.cc",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  parser.Parse(&argc, &argv);
+
+  EXPECT_EQ(1, argc);
+  EXPECT_STREQ("program", argv[0]);
+  EXPECT_FALSE(msg.has_rename());
+  EXPECT_TRUE(msg.has_build());
+  EXPECT_TRUE(msg.build().debug_mode());
+  EXPECT_STREQ("foo.cc", msg.build().file().c_str());
+}
+
+TEST(PositionalArguments, ProcessFileRenameSucceed) {
+  test::ProcessFile msg;
+
+  auto parser = stout::flags::Parser::Builder(msg).Build();
+
+  std::array arguments = {
+      "program",
+      "rename",
+      "foo.cc",
+      "bar.cc",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  parser.Parse(&argc, &argv);
+
+  EXPECT_EQ(1, argc);
+  EXPECT_STREQ("program", argv[0]);
+  EXPECT_TRUE(msg.has_rename());
+  EXPECT_FALSE(msg.has_build());
+  EXPECT_STREQ("foo.cc", msg.rename().cur_file_name().c_str());
+  EXPECT_STREQ("bar.cc", msg.rename().new_file_name().c_str());
+}
+
+TEST(PositionalArguments, BuildFileFailOnMissingRequiredArgument) {
+  test::ProcessFile msg;
+
+  auto parser = stout::flags::Parser::Builder(msg).Build();
+
+  std::array arguments = {
+      "program",
+      "build",
+      "--debug=true",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  EXPECT_DEATH(
+      parser.Parse(&argc, &argv),
+      "build: Failed while parsing and validating flags:"
+      "\?\n\?\n"
+      ". Argument 'file' .aka 'file_name'. not parsed but required");
+}
+
+TEST(PositionalArguments, RedundantPositionalArguments) {
+  test::ProcessFile msg;
+
+  auto parser = stout::flags::Parser::Builder(msg).Build();
+
+  std::array arguments = {
+      "program",
+      "build",
+      "main.cc",
+      "--debug=true",
+      "45",
+      "redundant",
+      "true",
+  };
+
+  int argc = arguments.size();
+  const char** argv = arguments.data();
+
+  EXPECT_DEATH(
+      parser.Parse(&argc, &argv),
+      "build: Failed while parsing and validating flags:"
+      "\?\n\?\n"
+      ". Encountered unknown flag '45'"
+      "\?\n\?\n"
+      ". Encountered unknown flag 'redundant'"
+      "\?\n\?\n"
+      ". Encountered unknown flag 'true'");
+}
+
+TEST(PositionalArguments, IllegalPositionalArgument1) {
+  test::IllegalPositionalArg1 msg;
+
+  EXPECT_DEATH(
+      stout::flags::Parser::Builder(msg).Build(),
+      "Field 'test.IllegalPositionalArg1.num' with 'stout::v1::argument'"
+      " extension must have string type");
+}
+
+TEST(PositionalArguments, IllegalPositionalArgument2) {
+  test::IllegalPositionalArg2 msg;
+
+  EXPECT_DEATH(
+      stout::flags::Parser::Builder(msg).Build(),
+      "Field 'test.IllegalPositionalArg2.str' should have option"
```

Comment:
Should we also test an invalid setup where we ~skip positions (e.g. there's only one positional argument, but its index is `5`)?